server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -349,6 +351,11 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
             while (iterator.hasNext()) {
                 Map.Entry<String, Object> entry = iterator.next();
                 String fieldName = entry.getKey();
+                Pattern p = Pattern.compile("[^a-zA-Z0-9_.@ $-]");
```

Comment:
Compile the pattern outside of the iteration?

Reply:
I'll add that now